wakebox/main.cpp

```diff
@@ -124,3 +188,90 @@ int main(int argc, char *argv[])
 
 	return res;
 }
+
+int main(int argc, char *argv[])
+{
+	unsigned int max_pairs = argc / 2;
+	char *tools[max_pairs];
+	char *binds[max_pairs];
+
+	struct option options[] {
+		{'r', "rootfs",       GOPT_ARGUMENT_REQUIRED},
+		{'t', "toolchain",    GOPT_ARGUMENT_REQUIRED | GOPT_REPEATABLE_VALUE, tools, max_pairs},
+		{'b', "bind",         GOPT_ARGUMENT_REQUIRED | GOPT_REPEATABLE_VALUE, binds, max_pairs},
+		{'x', "bind-cwd",     GOPT_ARGUMENT_FORBIDDEN},
+
+		{'p', "params",       GOPT_ARGUMENT_REQUIRED},
+		{'o', "output-stats", GOPT_ARGUMENT_REQUIRED},
+		{'i', "interactive",  GOPT_ARGUMENT_FORBIDDEN},
+		{'s', "force-shell",  GOPT_ARGUMENT_FORBIDDEN},
+
+		{'h', "help",         GOPT_ARGUMENT_FORBIDDEN},
+		{0,   0,              GOPT_LAST }
+	};
+
+	argc = gopt(argv, options);
+	gopt_errors("wakebox", options);
+
+	bool has_help = arg(options, "help")->count > 0;
+	bool has_params_file = arg(options, "params")->count > 0;
+	bool has_positional_cmd = argc > 1;
+
+	if (has_help) {
+		print_help();
+		return 1;
+	}
+
+	if (has_positional_cmd && has_params_file) {
+		std::cerr
+			<< "The batch mode --params argument can't be used with the interactive"
+                           " mode command argument."
+			<< std::endl;
+		return 1;
+	}
+
+	if (has_positional_cmd) {
+		std::string rootfs;
+		if (arg(options, "rootfs")->count > 0)
+			rootfs = arg(options, "rootfs")->argument;
+
+		std::vector<std::string> toolchains;
+		for (unsigned int i = 0; i < arg(options, "toolchain")->count; i++)
+			toolchains.push_back(tools[i]);
+
+		std::vector<mount_op> bind_ops;
+		for (unsigned int i = 0; i < arg(options, "bind")->count; i++) {
+			const std::string s = binds[i];
+			std::string source = s.substr(0, s.find(":"));
+			std::string destination = s.substr(s.find(":")+1);
```

Comment:
I assume you tested the 'no :' case? I usually explicitly test the result of s.find.